tests/test_updater_delegation_graphs.py

```diff
@@ -0,0 +1,267 @@
+#!/usr/bin/env python
+
+# Copyright 2021, New York University and the TUF contributors
+# SPDX-License-Identifier: MIT OR Apache-2.0
+
+"""Test updating delegated targets roles with various
+delegation hierarchies"""
+
+import os
+import sys
+import tempfile
+import unittest
+from dataclasses import astuple, dataclass, field
+from typing import Iterable, List, Optional
+
+from tests import utils
+from tests.repository_simulator import RepositorySimulator
+from tuf.api.metadata import (
+    SPECIFICATION_VERSION,
+    TOP_LEVEL_ROLE_NAMES,
+    DelegatedRole,
+    Targets,
+)
+from tuf.ngclient import Updater
+
+
+@dataclass
+class TestDelegation:
+    delegator: str
+    rolename: str
+    keyids: List[str] = field(default_factory=list)
+    threshold: int = 1
+    terminating: bool = False
+    paths: List[str] = field(default_factory=lambda: ["*"])
```

Comment:
nit: when you want to actually test path_hash_prefixes this needs to be optional... but we can fix that then